contracts/tokens/BaseToken.sol

```diff
@@ -153,5 +154,62 @@ abstract contract BaseToken is Context, ERC721Enumerable, CommunityOwnable {
         super._beforeTokenTransfer(from, to, firstTokenId, batchSize);
     }
 
+    /**
+     * @notice Batch transfers tokens by passing an array of token IDs.
+     * @dev This is simply delegates to `safeTranferFrom` for each token ID in the array.
+     */
+    function safeBatchTransferFrom(
+        address from,
+        address[] calldata receivers,
+        uint256[] calldata ids,
+        bytes memory data
+    )
+        public
+        virtual
+    {
+        _safeBatchTransferFrom(from, receivers, ids, data);
+    }
+
+    /**
+     * @notice Overloaded `safeBatchTransferFrom` that takes a single receiver
+     * address and an array of token IDs.
+     * @dev Same as `safeBatchTransferFrom` but this function prepares a `receivers`
+     * array with the same address.
+     */
+    function safeBatchTransferFrom(
+        address from,
+        address to,
+        uint256[] calldata ids,
+        bytes memory data
+    )
+        public
+        virtual
+    {
+        address[] memory receivers = new address[](ids.length);
+        uint256 i = 0;
+        for (; i < ids.length; i++) {
+            receivers[i] = to;
+        }
+        _safeBatchTransferFrom(from, receivers, ids, data);
+    }
+
+    function _safeBatchTransferFrom(
+        address from,
+        address[] memory receivers,
+        uint256[] memory ids,
+        bytes memory data
+    )
+        internal
+    {
+        if (receivers.length != ids.length) {
+            revert BaseToken_ReceiversAndIdsMismatch();
+        }
+
+        for (uint256 i = 0; i < ids.length; ++i) {
+            address to = receivers[i];
+            uint256 id = ids[i];
+            safeTransferFrom(from, to, id, data);
```

Comment:
we might avoid the 2 variables and just use `array[i]` in the parameters, but I leave it up to you, cause maybe it's not a big change in gas